### ISSUE TITLE
Parallelized Predictions

### ISSUE THREAD

**Issue by mthmn20** [maxpumperla/elephas#54]
Hi

It looks like the current predict method of a SparkModel doesn't run in a parallelized fashion. Is this indeed the case and, if so, are there plans to add that functionality?

Thanks

**Comment by maxpumperla** [comment]
It really depends on your application to see if that is really worth it. The point of elephas (for me) was always to do distributed training and get the resulting model back on master to work with it as regular keras model. Usually predictions are relatively fast and one shouldn't underestimate the overhead it takes to construct e.g. an RDD from test data. Often this already outweighs the speed gains for parallel predictions.

But in general you're right, this is certainly an interesting suggestion and I will think about it for future releases. Thanks!

**Comment by mthmn20** [comment]
Thanks! I agree that parallelized prediction probably wouldn't be very useful for production systems, the use case I have in mind is for training when I want to calculate different metrics, some of which can't be implemented in Theano/Keras (or at least not at all straightforwardly), on validation data between epochs, and then do things like checkpoint the model at the epochs it does best on each metric. 

Related question: if you pass a validation_split argument to train, that validation will happen in parallel, right?

**Comment by aditsanghvi94** [comment]
Any update on whether parallelized predictions are in the pipeline or an easy way to do them?

**Comment by danielenricocahall** [comment]
I just finished implementing distributed predict and evaluate - they should be in the next release coming out this week!

**Comment by danielenricocahall** [comment]
Officially added support for parallel prediction and evaluation in 1.1.0: https://github.com/maxpumperla/elephas/releases/tag/1.1.0 feel free to play with it and let me know how it goes!
